ResearchKit/SurveyTasks/ORKStrokeImpactScale.m

```diff
@@ -0,0 +1,193 @@
+    
+    for (int questionID=0; questionID<59; questionID++) {
+        ORKQuestionStep *step = [[ORKQuestionStep alloc] initWithIdentifier:[NSString stringWithFormat:@"%d",questionID]];
+        switch (questionID) {
+            case 0:
+            case 1:
+            case 2:
+            case 3:
+                step.answerFormat = answerStrengthFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_PHYSICAL", nil);
+                break;
+            case 4:
+            case 5:
+            case 6:
+            case 7:
+            case 8:
+            case 9:
+            case 10:
+                step.answerFormat = answerDifficultyFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_MEMORY", nil);
+                break;
+            case 11:
+            case 12:
+            case 13:
+            case 14:
+            case 15:
+            case 16:
+            case 17:
+            case 18:
+            case 19:
+                step.answerFormat = answerTimeFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_MOOD", nil);
+                break;
+            case 20:
+            case 21:
+            case 22:
+            case 23:
+            case 24:
+            case 25:
+            case 26:
+                step.answerFormat = answerDifficultyFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_COMMUNICATION", nil);
+                break;
+            case 27:
+            case 28:
+            case 29:
+            case 30:
+            case 31:
+            case 32:
+            case 33:
+            case 34:
+            case 35:
+            case 36:
+                step.answerFormat = answerDifficultyFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_ACTIVITIES", nil);
+                break;
+            case 37:
+            case 38:
+            case 39:
+            case 40:
+            case 41:
+            case 42:
+            case 43:
+            case 44:
+            case 45:
+                step.answerFormat = answerDifficultyFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_MOBILITY", nil);
+                break;
+            case 46:
+            case 47:
+            case 48:
+            case 49:
+            case 50:
+                step.answerFormat = answerDifficultyFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_HAND", nil);
+                break;
+            default:
+                step.answerFormat = answerTimeFormat;
+                step.title = ORKLocalizedString(@"SIS_TITLE_PARTICIPATE", nil);
+        }
+        {
+            NSString *questionStem = [NSString stringWithFormat:@"SIS_Q%d",questionID];
+            step.text = ORKLocalizedString(questionStem, nil);
+        }
+        ORKStepArrayAddStep(steps,step);
```

Comment:
I had thought about this for some time. My concern was that with more significant strokes, cognitive processing of symbols and actions can be much harder, and the scale tools have a surprising amount of implied action about them. We do have to be cautious not to cognitively overburden neuro-rehab patients.

So in terms of UI my feel was making each option explicit and being a 'touch the option' rather than 'move the scale' was cognitively simpler, albeit more verbose. If I were doing it as an assessor tool for carers or professionals, I would probably choose the approach above. Notice you also have to make some cognitive shortcuts as to what is between the two bars. The last page of the Stroke Impact scale still has a vertical slider but that bit of the test is not particularly well-validated compared to the other domains.

I'm also conscious that neuro injury patients would benefit more from lots of small steps rather than fewer big steps (this is why there are multiple scales testing different elements as well - in Real Life we tend to pick and mix the tool according to the patient: many couldn't manage the Stroke Impact Scale for example, or you need tools focussing on the really simple. I have two more tools I've not added yet - the Rankin scale and the Barthel Index. These tend to focus on much more basic concepts still. In practice if you have a high Rankin score (or a low Barthel), you wouldn't ask the more complex tasks.